client/src/bundles/UserForms.js

```diff
@@ -1,738 +1,742 @@
-				onfocusout: false
-			});
-		}
-
-		if (CONSTANTS.DISPLAY_ERROR_MESSAGES_AT_TOP) {
-			$.extend(UserForm.prototype.validationOptions, {
-				// Callback for custom code when an invalid form / step is submitted.
-				invalidHandler: function (event, validator) {
-					$userform.trigger('userform.form.error', [validator]);
-				},
-				onfocusout: false
-			});
-		}
-
-		// Display all the things that are hidden when JavaScript is disabled.
-		$('.userform-progress, .step-navigation').attr('aria-hidden', false).show();
-
-		// Extend classes with common functionality.
-		$.extend(FormStep.prototype, commonMixin);
-		$.extend(ErrorContainer.prototype, commonMixin);
-
-		userform = new UserForm($userform);
-
-		// Conditionally hide field labels and use HTML5 placeholder instead.
-		if (CONSTANTS.HIDE_FIELD_LABELS) {
-			$userform.find('label.left').each(function () {
-				var $label = $(this);
-
-				$('[name="' + $label.attr('for') + '"]').attr('placeholder', $label.text());
-				$label.remove();
-			});
-		}
-
-		// Initialise the form steps.
-		userform.$el.find('.form-step').each(function (i, element) {
-			var step = new FormStep(element);
-
-			userform.addStep(step);
-		});
-
-		userform.setCurrentStep(userform.steps[0]);
-
-		// Initialise actions and progressbar
-		progressBar = new ProgressBar($('#userform-progress'));
-		formActions = new FormActions($('#step-navigation'));
-
-		// Enable jQuery UI datepickers
-		$(document).on('click', 'input.text[data-showcalendar]', function() {
-			var $element = $(this);
-
-			$element.ssDatepicker();
-
-			if($element.data('datepicker')) {
-				$element.datepicker('show');
-			}
-		});
-
-		// Make sure the form doesn't expire on the user. Pings every 3 mins.
-		setInterval(function () {
-			$.ajax({ url: 'UserDefinedForm_Controller/ping' });
-		}, 180 * 1000);
-
-		// Bind a confirmation message when navigating away from a partially completed form.
-		var form = $('form.userform');
-		if(typeof form.areYouSure != 'undefined') {
-			form.areYouSure({
-				message: ss.i18n._t('UserForms.LEAVE_CONFIRMATION', 'You have unsaved changes!')
-			});
-		}
-	}
-
-	main();
+
+import $ from 'jquery';
+
+$(document).ready(() => {
+  // A reference to the UserForm instance.
+  let userform = null;
+
+  // Settings that come from the CMS.
+  const CONSTANTS = {};
+
+  // Common functions that extend multiple classes.
+  const commonMixin = {
+    /**
+     * @func show
+     * @desc Show the form step. Looks after aria attributes too.
+     */
+    show: () => {
+      this.$el.attr('aria-hidden', false).show();
+    },
+    /**
+     * @func hide
+     * @desc Hide the form step. Looks after aria attributes too.
+     */
+    hide: () => {
+      this.$el.attr('aria-hidden', true).hide();
+    },
+  };
+
+  /**
+   * @func UserForm
+   * @constructor
+   * @param {object} element
+   * @return {object} - The UserForm instance.
+   * @desc The form
+   */
+  function UserForm(element) {
```

Comment:
Move this function (plus the `UserForm.prototype.*` calls) to just above `function main() {` :)